src/codegen/compile.ml

```diff
@@ -8296,19 +8691,32 @@ module StackRep = struct
     | UnboxedTuple n, Vanilla -> Tuple.from_stack env n
     | Vanilla, UnboxedTuple n -> Tuple.to_stack env n
 
-    | UnboxedWord64, Vanilla -> BoxedWord64.box env
-    | Vanilla, UnboxedWord64 -> BoxedWord64.unbox env
+    | UnboxedWord64 pty, Vanilla -> BoxedWord64.box env pty (* ! *)
+    | Vanilla, UnboxedWord64 pty -> BoxedWord64.unbox env pty (* ! *)
+
+    (* TaggedSmallWord types *)
+    | UnboxedWord32 (Type.(Int8 | Nat8 | Int16 | Nat16 | Char) as pty), Vanilla ->
+       TaggedSmallWord.tag env pty
+    | Vanilla, UnboxedWord32 (Type.(Nat8 | Int8 | Nat16 | Int16 | Char) as pty) ->
+       TaggedSmallWord.untag env pty
 
-    | UnboxedWord32, Vanilla -> BoxedSmallWord.box env
-    | Vanilla, UnboxedWord32 -> BoxedSmallWord.unbox env
+    (* BoxedSmallWord types *)
+    (* TODO: constrain pty *)
```

Comment:
This would be `(Type.(Int32 | Nat32) as pty)`?

Reply:
Added an assert instead (also for BoxedWord64)